lib/puppet-lint/plugins/check_parameter_documentation.rb

```diff
@@ -11,7 +11,7 @@ def check
           if dtok.value =~ /\A\s*\[\*([a-zA-Z0-9_]+)\*\]/ or
              dtok.value =~ /\A\s*\$([a-zA-Z0-9_]+):: +/ or
              dtok.value =~ /\A\s*@param (?:\[.+\] )?([a-zA-Z0-9_]+)(?: +|$)/
-            doc_params << $1
+            doc_params.unshift( { value: $1, token: dtok }) # Push new params onto start of array
```

Comment:
I was thinking about `Hash doc_params`. If the param already exists, add it to `Array duplicates`.

You may want to consider also adding a warning on the line with the original. This can make it easy to find the pair of duplicates and merge them. However, not strictly needed and it's probably easy enough if you don't.

Reply:
I was wondering why you wanted a hash, but I guess that'll be needed for fixing some of the other issues I opened.